metrics-operator/api/v1alpha3/analysisdefinition_types.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AnalysisDefinitionSpec defines the desired state of AnalysisDefinition
+type AnalysisDefinitionSpec struct {
+	// Objectives defines a list of objectives for evaluation
+	Objectives []Objective `json:"objectives,omitempty"`
+	// TotalScore defines the required score for an evaluation to be successful
+	TotalScore Score `json:"totalScore"`
+}
+
+// Score defines the required score for an evaluation to be successful
+type Score struct {
+	// PassPercentage defines the threshold which needs to be reached for an evaluation to pass.
+	PassPercentage int `json:"passPercentage"`
+	// WarningPercentage defines the threshold which needs to be reached for an evaluation to pass with a 'warning' status.
+	WarningPercentage int `json:"warningPercentage"`
+}
+
+// Objective defines a list of objectives
+type Objective struct {
+	// AnalysisValueTemplateRef defines a reference to the used AnalysisValueTemplate
+	AnalysisValueTemplateRef ObjectReference `json:"analysisValueTemplateRef"`
+	// SLOTargets defines a list of SLOTargests
+	SLOTargets SLOTarget `json:"sloTargets,omitempty"`
```

Comment:
This is a little bit weird, but `SLOTargets SLOTarget` makes the most sense here. It's because we specify multiple pass and warning targets. Therefore plural in the naming is the right choice here. Also, on this level we can specify only pass and warning targets, where the targeting details are one level down in `Pass` or `Warning` parameters. To be honest, I would leave it as it is